reports/historical.py

```diff
@@ -147,9 +147,13 @@
 
 
 def canon(key):
-    key = key.replace(u'\u2026', '')
-    key = recode(key.strip())
-    return key.strip().lower()
+    print(key)
+    try:
+        key = key.replace(u'\u2026', '')
+        key = recode(key.strip())
+        return key.strip().lower()
+    except AttributeError:
+        return ''
```

Comment:
What is happening here?

Reply:
When the columns are empty cells, `key` becomes empty and `key = key.replace(u'\u2026', '')` throws an `AttributeError` which causes the program to exit halfway. This is why I made these changes @kilemensi

Reply:
1. Assuming what you're saying is correct @ascii-dev , shouldn't you just do `if not key:` to handle it? Why do we need to be dealing with exceptions?
2. If key is empty i.e. empty cells, that could also mean we're reading the wrong thing as well.

Reply:
> 2. If key is empty i.e. empty cells, that could also mean we're reading the wrong thing as well.

There are empty cells in between the data itself @kilemensi 
![image](https://user-images.githubusercontent.com/8318974/101146835-cb6f0680-361b-11eb-8a31-60615b3faab2.png)

Reply:
> 1. Assuming what you're saying is correct @ascii-dev , shouldn't you just do `if not key:` to handle it? Why do we need to be dealing with exceptions?

This also works. `try...except` was the first thing that came to mind.